base/client.jl

```diff
@@ -71,6 +71,18 @@ function repl_cmd(cmd, out)
         ENV["OLDPWD"] = new_oldpwd
         println(out, pwd())
     else
+        @windows_only begin
+            # build correctly quoted string for cmd.exe
+            special = r".*[ &()[\]{}\^=;!'+,`~]+.*"
```

Comment:
- I used the list of characters that need to be quoted in filenames according to `cmd /?`, i.e.
```
The special characters that require quotes are:
     <space>
     &()[]{}^=;!'+,`~
```
`"` is missing because it's not allowed in filenames.

The current code looks for special characters and surrounds the corresponding arguments by `"`. This works well for filenames, but not necessarily for switches containing special characters.

I think one root problem is that `shell_parse` swallows unquoted `"`s 
```
julia> `foo --bar="my option" "my file"`
`foo '--bar=my option' 'my file'`

julia> ans.exec
3-element Array{Union(UTF8String,ASCIIString),1}:
 "foo"
 "--bar=my option"
 "my file"
```
so we can't just pass the command verbatim to spawn without messing with the `"`s. And of course we don't really know which arguments are files and which ones are options.

- What do you mean by `single linear pass` in this context?